xapian-core/include/xapian/weight.h

```diff
@@ -1203,28 +1197,19 @@ class XAPIAN_VISIBILITY_DEFAULT LMWeight : public Weight {
      *					DIRCHLET(2000))
      *
      *  @param param_smoothing2_	A non-negative parameter which is used
-     *					only with TWO_STAGE_SMOOTHING as
-     *					parameter for Dirichlet's smoothing.
-     *					(default: 2000)
-     *
-     *  @param param_delta_	A non-negative parameter for pseudo tf value to control the scale
-     *		      		of the tf lower bound in the Dir+ scoring funtion. Delta can be tuned
-     *				from 0.0 to 0.15 in increments of 0.01 but it is also advisable to
-     *				fix it to 0.05 for normal use. (default 0.05)
+     *					with TWO_STAGE_SMOOTHING as parameter for Dirichlet's
+     *					smoothing (default: 2000) and as parameter delta to
+     *					control the scale of the tf lower bound in the
+     *             			DIRICHLET_PLUS_SMOOTHING (default 0.05).
      *
-     *  @param enable_dirplus_	A boolean parameter to control the weight calculations. It takes
-     *				effect only when smoothing is set to Dirichlet. By default, it is set to
-     *				zero so as to allow normal functioning of Dirichlet smoothing. (default 0)
      */
     // Unigram LM Constructor to specifically mention all parameters for handling negative log value and smoothing.
     explicit LMWeight(double param_log_ = 0.0,
 		      type_smoothing select_smoothing_ = TWO_STAGE_SMOOTHING,
 		      double param_smoothing1_ = 0.7,
-		      double param_smoothing2_ = 2000.0,
-		      double param_delta_ = 0.05,
-		      bool enable_dirplus_ = 0)
+		      double param_smoothing2_ = 2000.0)
```

Comment:
I think it's better to handle it just below here, where we initialise `param_smoothing2` - that means the `param_smoothing2` member always has a valid value.  If we handle it in `LMWeight::init()`, we no longer have the invariant `param_smoothing2 >= 0`.

I think the other similar things which currently get handled in `init()` would be better done here instead (and we should use `-1.0` as the default value for `param_smoothing1_` too - currently if you ask for ` DIRICHLET_SMOOTHING` and explicitly set `smoothing1` to `0.7`, that code can't tell that was explicit rather than the default value and changes it to `2000.0` which is nasty).  The `get_query_length()` check can't be done here though.  But let's leave those changes until after we've got Dir+ merged.